olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -431,13 +472,19 @@ DefaultCacheImpl::GetFromDiscCache(const std::string& key) {
   if (mutable_cache_) {
     auto expiry = GetRemainingExpiryTime(key, *mutable_cache_);
     if (expiry <= 0) {
-      PurgeDiskItem(key, *mutable_cache_);
-      if (mutable_cache_lru_) {
-        mutable_cache_lru_->Erase(key);
+      const auto result = mutable_cache_->Get(key);
+      if (result) {
+        mutable_cache_data_size_ -= key.size() + result.value().size();
       }
+      mutable_cache_data_size_ -= GetExpirySize(key, *mutable_cache_);
+
+      PurgeDiskItem(key, *mutable_cache_);
+      RemoveKeyLru(key);
     } else {
       if (!PromoteKeyLru(key)) {
         // If not found in LRU no need to look in disk cache either.
+        OLP_SDK_LOG_TRACE_F(
```

Comment:
Trace level should only be used for method entry and exit. Debug is for the rest, which should not be visible in production.